azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/Key.java

```diff
@@ -16,44 +16,136 @@
  */
 package org.jclouds.azurecompute.arm.domain;
 
-import java.util.Date;
+import java.util.Map;
+import java.util.List;
 
 import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.Json;
 import org.jclouds.json.SerializedNames;
 
 import com.google.auto.value.AutoValue;
 
 @AutoValue
 public abstract class Key {
+   @AutoValue
+   public abstract static class JsonWebKey {
+      @Nullable
+      public abstract String crv();
+      @Nullable
+      public abstract String d();
+      @Nullable
+      public abstract String dp();
+      @Nullable
+      public abstract String dq();
+      @Nullable
+      public abstract String e();
+      @Nullable
+      public abstract String k();
+      @Nullable
+      public abstract String key_hsm();
```

Comment:
I think we generally prefer camel notation for fields (keyHsm) and use the json name into serializedNames